skyportal/handlers/api/spectrum.py

```diff
@@ -72,39 +70,22 @@ def post(self):
                 f'Invalid / missing parameters; {e.normalized_messages()}'
             )
 
-        group_ids = data.pop("group_ids")
-        if group_ids == "all":
-            groups = (
-                DBSession()
-                .query(Group)
-                .filter(Group.name == cfg["misc"]["public_group_name"])
-                .all()
-            )
+        # always append the single user group
+        single_user_group = self.associated_user_object.single_user_group
+
+        group_ids = data.pop("group_ids", None)
+        if not group_ids:
```

Comment:
```suggestion
        if group_ids is None:
```
(more idiomatic in Python)